### PR TITLE
Add include_nodes filter for jenkins

### LINKED ISSUE #9047: support include nodes for jenkins plugin
## Feature Request

This feature is requested for Jenkins input plugin 
https://github.com/influxdata/telegraf/tree/master/plugins/inputs/jenkins 

### Proposal:

As you could have ephemeral agent with Kuberntenes in Jenkins, there is no way if you have to filter them or exclude them from list of normal agents.
would be awesome if we could have node_include and mix it with node_exclude like
node_include = ["nodeA", "nodeB"]
node_exclude = [ "*" ]
so just filter the nodes we specify in the configuration. 

### Current behavior:

There is no node_include in jenkins input plugin. 

### Desired behavior:

Add ability to add specific Jenkins agent in Jenkins input plugin and exclude everything else. 

### Use case: 

Many people now use mix jenkins normal agent and ephemeral agent with kubernetes, and there is no other way to filter agents node in current jenkins input plugin. 

Regards,

### PR DESCRIPTION
**Pull request by akrantz01**
### Required for all PRs:

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.

resolves #9047 

Adds the ability to explicitly include nodes for the `jenkins_node` metric. Previously, you could only exclude nodes from being collected.

Like the `job_exclude` and `job_include` filters, `node_exclude` takes precedence over `node_include` if they are both present.

### CODE REVIEW

**Comment by akrantz01** [comment]
@G0lang Would you be willing to test this?

**Review by srebhan — CHANGES_REQUESTED** [review]
@akrantz01 thanks for this PR, it makes totally sense. However, you probably should use `NewIncludeExcludeFilter()` of the filter package which does the include/exclude handling for you and reduces the amount of code.

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks for also tackling `jobFilter`! I have just one minor comment in the code, but it is optional.

File: plugins/inputs/jenkins/jenkins.go [hunk 1]
```diff
@@ -203,8 +198,9 @@ func (j *Jenkins) gatherNodeData(n node, acc telegraf.Accumulator) error {
 	}
 
 	tags["node_name"] = n.DisplayName
-	// filter out excluded node_name
-	if j.nodeFilter != nil && j.nodeFilter.Match(tags["node_name"]) {
+
+	// filter out excluded or not included node_name
+	if j.nodeFilter != nil && !j.nodeFilter.Match(tags["node_name"]) {
```
[the remark this answers is not in the document]
Comment (akrantz01) [reply plugins/inputs/jenkins/jenkins.go]: Oh, you're right; I didn't even realize.

**Review by srebhan — APPROVED** [review]
Perfect, thank you! LGTM.

**Review by MyaLongmire — APPROVED** [review]

**Pull request merged**